greenplumpython/table.py

```diff
@@ -52,6 +52,26 @@ def select(self, target_list: Iterable) -> "Table":
             parents=[self],
         )
 
+    def join(
+        self,
+        other: "Table",
+        expr: expr.Expr,
+        how: Optional[str] = "natural",
```

Comment:
I think it is more clear to support multiple JOIN types with different methods, e.g., `outer_join()`, `cross_join()`, etc., insead of using the `how` argument here.

Reply:
If we separate them don't we have repeated lines of code?

Reply:
We can have a private function that do the real job, and the public functions do nothing but call the private function

Reply:
Got it